### ISSUE TITLE
Multiply Primitive should support boolean variable type

### ISSUE THREAD

**Issue by kmax12** [Featuretools/featuretools#752]
Currently, the input to the multiply primitive must be numeric. It is common when creating seed features to want to multiple a numeric value and a boolean (0/1) together. For example, look at this [stackoverflow post](https://stackoverflow.com/questions/57997126/how-to-write-seed-features-that-include-a-conditional-statement?noredirect=1#comment102409311_57997126).

The follow code below should work

```python
import featuretools as ft
import pandas as pd

es = ft.EntitySet()

df = pd.DataFrame({"bool": [True, False], "numeric": [1, 1]})

es.entity_from_dataframe(entity_id="test",
                         dataframe=df)

ft.Feature(es["test"]["bool"]) * ft.Feature(es["test"]["numeric"])
```

It currently gives this error

```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-5-2558a225064f> in <module>
      9                          dataframe=df)
     10 
---> 11 ft.Feature(es["test"]["bool"]) * ft.Feature(es["test"]["numeric"])

~/Documents/featuretools/featuretools/feature_base/feature_base.py in __mul__(self, other)
    287     def __mul__(self, other):
    288         """Multiply by other"""
--> 289         return self._handle_binary_comparision(other, primitives.MultiplyNumeric, primitives.MultiplyNumericScalar)
    290 
    291     def __rmul__(self, other):

~/Documents/featuretools/featuretools/feature_base/feature_base.py in _handle_binary_comparision(self, other, Primitive, PrimitiveScalar)
    230     def _handle_binary_comparision(self, other, Primitive, PrimitiveScalar):
    231         if isinstance(other, FeatureBase):
--> 232             return Feature([self, other], primitive=Primitive)
    233 
    234         return Feature([self], primitive=PrimitiveScalar(other))

~/Documents/featuretools/featuretools/feature_base/feature_base.py in __new__(self, base, entity, groupby, parent_entity, primitive, use_previous, where)
    755                                                primitive=primitive,
    756                                                groupby=groupby)
--> 757             return TransformFeature(base, primitive=primitive)
    758 
    759         raise Exception("Unrecognized feature initialization")

~/Documents/featuretools/featuretools/feature_base/feature_base.py in __init__(self, base_features, primitive, name)
    660                                                relationship_path=RelationshipPath([]),
    661                                                primitive=primitive,
--> 662                                                name=name)
    663 
    664     @classmethod

~/Documents/featuretools/featuretools/feature_base/feature_base.py in __init__(self, entity, base_features, relationship_path, primitive, name, names)
     56         self._names = names
     57 
---> 58         assert self._check_input_types(), ("Provided inputs don't match input "
     59                                            "type requirements")
     60 

AssertionError: Provided inputs don't match input type requirements
```

**Comment by thehomebrewnerd** [comment]
@kmax12 Do you want the ability to multiple by a boolean by a scalar as well? 

Something like this: `ft.Feature(es["test"]["bool"]) * 3`

**Comment by kmax12** [comment]
@thehomebrewnerd I think we can hold off on the ability to multiple a boolean by a scalar right now. im not sure of a good use case for it, but if one comes up we can always add.

**Comment by thehomebrewnerd** [comment]
Closed via PR #756
